js/src/Components/Widgets/CkeditorWidget.js

```diff
@@ -29,30 +29,40 @@ export class CkeditorWidget extends Component {
     this.showHideSummary = this.showHideSummary.bind(this);
     this.unlockSnapshot = this.unlockSnapshot.bind(this);
     this.getSummaryInput = this.getSummaryInput.bind(this);
+    this.addListeners = this.addListeners.bind(this);
   }
 
   componentDidMount() {
     try {
+      CKEDITOR.on('instanceReady', this.addListeners);
       Drupal.behaviors.editor.attach(this.widgetRef.current, window.drupalSettings);
 
       Object.keys(Drupal.editors).map(editorId => {
         Drupal.editors[editorId].onChange(this.textareaRef, this.onEditorChange.bind(undefined, this.textareaRef));
-        CKEDITOR.instances[`${this.props.fieldId}-text-area`].on('unlockSnapshot', this.unlockSnapshot.bind(undefined, this.textareaRef));
       })
     }
     catch (error) {
     }
   }
 
+  addListeners() {
+    CKEDITOR.instances[`${this.props.fieldId}-text-area`].on('unlockSnapshot', this.unlockSnapshot.bind(undefined, this.textareaRef));
+    CKEDITOR.instances[`${this.props.fieldId}-text-area`].on("key", this.unlockSnapshot.bind(undefined, this.textareaRef));
+  }
+
   unlockSnapshot(element, snapshot) {
-    if (snapshot.editor.getData().length >= 1) {
-      this.onEditorChange(element, snapshot.editor.getData());
-    }
+    // Timeout to let the keyed in value to be added to the data.
+    setTimeout(() => {
```

Comment:
Is the timeout because events are happening asynchronously? I'll let this pass for now but this has a code smell that should likely be looked in to.

Reply:
Interestingly the timeout is only really needed when "source" is enabled. There is a way to avoid this. Basically when the user clicks "source" creditor creates another `<textarea>` for entry. It is possible to add an `onChange` listener to that.

Reply:
Do you want to make that change to this PR or create a new ticket to revise this later?

Reply:
Let's create another ticket. At this moment I dont know exactly how to target that source mode text area

Reply:
I'll merge this but please post a link to the ticket in this thread somewhere.